src/main/kotlin/app/batch/DecompressionProcessor.kt

```diff
@@ -0,0 +1,36 @@
+package app.batch
+
+import org.apache.commons.compress.compressors.CompressorStreamFactory
+import org.apache.commons.compress.compressors.gzip.GzipCompressorInputStream
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.batch.item.ItemProcessor
+import org.springframework.stereotype.Component
+import java.io.InputStream
+
+@Component
+class DecompressionProcessor : ItemProcessor<DecryptedStream, InputStream> {
+
+    override fun process(item: DecryptedStream): InputStream? {
+        val inputStream = item.inputStream
+        val fileName = item.fileName
+        try {
+            val compressorInputStream = CompressorStreamFactory().createCompressorInputStream(CompressorStreamFactory.GZIP,
+                inputStream) as GzipCompressorInputStream
+            logger.info("Compressed size of the file $fileName : ${compressorInputStream.compressedCount}")
```

Comment:
Is it worth adding to the log here that the file is about to be processed? Maybe appending `... Now decompressing` or something

Reply:
Seconded @benj-fletch

Reply:
Added the logger before start of decompression